src/Activity.js

```diff
@@ -1,31 +1,82 @@
-class Activity {
-  constructor(data, userRepository) {
-    this.userId = data.userID;
-    this.date = data.date;
-    this.steps = data.numSteps;
-    this.minutesActive = data.minutesActive;
-    this.flightsOfStairs = data.flightsOfStairs;
-    this.milesWalked = 0;
-    this.reachedStepGoal = null;
-    this.doActivity(userRepository);
-  }
-  doActivity(userRepo) {
-    var activity = this;
-    userRepo.users.find(function(user) {
-      return user.id === activity.userId;
-    }).updateActivities(this);
-  }
-  calculateMiles(userRepository) {
-    let walkingUser = userRepository.users.find(user => {
-      return user.id === this.userId;
+import User from './User';
+
+class Activity extends User {
+  constructor(userData, date) {
+    super(userData, date); 
+    this.steps = 0;
+    this.minutesActive = 0;
+    this.flightsOfStairs = 0;
+    this.weeklyAverageActive = 0; 
+    this.weeklyAverageSteps = 0; 
+    this.reachedStepGoal = false;
+    this.activityRecord = [];
+  }
+    
+  findTodayActivityData(activityData) {
+    const todaysData = activityData.find(activity => {
+      return this.id === activity.userID && this.date === activity.date;
+    }) 
+    this.steps = todaysData.numSteps;
+    this.minutesActive = todaysData.minutesActive; 
+    this.flightsOfStairs = todaysData.flightsOfStairs; 
+  }
+
+  updateActivities(activityData) {
+     activityData.forEach(activity => {
+      this.activityRecord.unshift(activity);
+    });
+  }
+  
+  calculateMiles() {
+    return Number(((this.steps * this.strideLength) / 5280).toFixed(1));
+  }
+
+  calcWeeklyAverageActive(date) {
+    const currentDateIndex = this.activityRecord.findIndex(activity => {
+      return activity.date === date; 
     });
-    return Math.round(this.steps * walkingUser.strideLength / 5280).toFixed(1);
+    const currentWeekData = this.activityRecord.slice(currentDateIndex, currentDateIndex + 7); 
+    const averageWeeklyData = currentWeekData.reduce((weeklyAverage, day) => {
+      if(!weeklyAverage.steps) {
+       weeklyAverage.steps = day.numSteps;
+       weeklyAverage.minutesActive = day.minutesActive; 
+    } else {
+       weeklyAverage.steps += day.numSteps;
+       weeklyAverage.minutesActive += day.minutesActive; 
+    }  
+       return weeklyAverage; 
+    }, {}); 
```

Comment:
This reduce is doing a lot to make it work as a reduce. Thoughts for refactoring:

`const averageWeeklyData = {steps: 0, minutesActive: 0};
    currentWeekData.forEach(day => {
      averageWeeklyData.steps += day.numSteps;
      averageWeeklyData.minutesActive += day.minutesActive;
    });`

Reply:
Oh nice, yeah I will implement this, I didn't like how big it was getting myself. Nice suggestion